pkg/cli/adbtunnel.go

```diff
@@ -368,16 +367,18 @@ func gatherADBTunnels(controlDir string, zone, host string) ([]ADBForwarderStatu
 	return forwarders, merr
 }
 
-// Finds an existing tunnel or creates a new one.
-// Returns the port for the existing or created tunnel, the tunnel controller if
-// one was created and/or an error. A negative port number indicates failure, errors
-// may still be returned on success and could be presented to the user as warnings.
-func findOrCreateTunnel(controlDir string, devSpec DeviceSpec, service client.Service) (int, *TunnelController, error) {
+type findOrCreateRet struct {
+	port        int
+	controller  *TunnelController
+	nonFatalErr error
```

Comment:
nit: could just `error` be fine here? the `nonFatal` is an implementation detail, users of this structure  are responsible of treating `t.error` as a non fatal error.

Reply:
Done